README.md

```diff
@@ -1,18 +1,352 @@
+    # This subclass overrides the bounds of the `retries` Parameter
+    # and the default of the `verbose` Parameter. All the other
+    # Parameter metadata are inherited from Processor.
+    retries = param.Integer(bounds=(0, 100))
+    verbose = param.Boolean(default=True)
+
+
+# Attributes exist at both the class-level and instance-level
+print(CustomProcessor.verbose)  # => True
+
+# The `.param` namespace is also available at the class-level.
+# Parameter metadata inheritance in action, with `default`
+# inherited from Processor and `bounds` overridden by CustomProcessor.
+print(CustomProcessor.param['retries'].default)  # => 3
+print(CustomProcessor.param['retries'].bounds)  # => (0, 100)
+
+# Class attributes are also runtime validated.
+try:
+    CustomProcessor.retries = 200
+except ValueError as e:
+    print(e)
+    # => Integer parameter 'CustomProcessor.retries' must be at most 100, not 200.
+
+cprocessor = CustomProcessor()
+
+# As in normal Python classes, class-level attribute values apply
+# to all instances that didn't override it.
+print(cprocessor.mode)  # => 'fast'
+Processor.mode = 'accurate'
+print(cprocessor.mode)  # => 'accurate'
+```
+
+## Reactive Programming
+
+Param extends beyond rich class attributes with a suite of APIs for reactive programming. Let's do a quick tour!
+
+We'll start with APIs that trigger side-effects only, which either have the name watch in their name or are invoked with `watch=True`:
+
+1. `<parameterized_obj>.param.watch(fn, *parameters, ...)`: Low-level, imperative API to attach callbacks to parameter changes, the callback receives one or more rich `Event` objects.
+2. `@depends(*parameter_names, watch=True)`: In a Parameterized class, declare dependencies and automatically watch parameters for changes to call the decorated method.
+3. `@depends(*parameters, watch=True)`: Equivalent of 2 above to decorate functions.
+4. `bind(fn, *references, watch=True, **kwargs)`: Function binding with automatic references (parameters, bound functions, reactive expressions) watching and triggering on changes.
+
+Running the snippet below, you will see that updating the parameters `a`, `b`, `c`, and `d` will emit print statements.
+
+```python
+import param
+
+class SideEffectExample(param.Parameterized):
+    a = param.String()
+    b = param.String()
+    c = param.String()
+    d = param.String()
+
+    def __init__(self, **params):
+        super().__init__(**params)
+        self.param.watch(lambda event: print(event), 'a')  # 1.
+
+    @param.depends('b', watch=True)  # 2.
+    def print_b(self):
+        print(f"print_b: {self.b=}")
+
+sfe = SideEffectExample()
+sfe.a = 'foo'
+# Event(what='value', name='a', obj=SideEffectExample(a='foo', b='', c='', d='', name='SideEffectExample00015'), cls=Example(a='foo', b='', c='', d='', name='SideEffectExample00015'), old='', new='foo', type='changed')
+
+sfe.b = 'bar'
+# print_b: self.b='bar'
+
+@param.depends(sfe.param.c, watch=True)  # 3.
```

Comment:
I would suggest not documenting using `depends()` outside of a class because of how the decorator syntax encourages early binding to a reactive object, which I think limits how the function can be used. I don't know of any advantage that usage has over `bind` and would always steer users to `bind` instead when using functions.